File: plugins/sem/tests/fixtures/roundtrip/go/basic.go
package fixtures

import "fmt"

type User struct {
	ID   string
	Name string
}

func Greeting(user User) string {
	return fmt.Sprintf("Hello, %s!", user.Name)
}

func IsEmpty(user User) bool {
	return user.ID == "" && user.Name == ""
}

File: plugins/sem/tests/fixtures/roundtrip/rust/basic.rs
use std::fmt;

// Keep this comment outside semantic entities.
const DEFAULT_NAME: &str = "Ada";

pub fn greeting(name: &str) -> String {
    format!("Hello, {name}!")
}

pub struct Greeter {
    name: String,
}

impl fmt::Display for Greeter {
    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
        formatter.write_str(&greeting(&self.name))
    }
}

// Tail gap should survive rendering.
